be-traveltrove/src/main/java/com/traveltrove/betraveltrove/presentation/events/EventController.java

```diff
@@ -0,0 +1,82 @@
+package com.traveltrove.betraveltrove.presentation.events;
+
+import com.traveltrove.betraveltrove.business.event.EventService;
+import com.traveltrove.betraveltrove.utils.exceptions.NotFoundException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+@RestController
+@RequestMapping("/api/v1/events")
+@Slf4j
+@CrossOrigin(origins = "http://localhost:3000")
+public class EventController {
+
+    private final EventService eventService;
+
+    public EventController(EventService eventService) {
+        this.eventService = eventService;
+    }
+
+    // Get all events
+    @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
+    public Flux<EventResponseModel> getAllEvents() {
+        log.info("Fetching all events");
+        return eventService.getEvents();
+    }
+
+    // Get all events by city ID
+    @GetMapping(value = "/city/{cityId}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public Flux<EventResponseModel> getEventsByCityId(@PathVariable String cityId) {
+        log.info("Fetching all events by city ID: {}", cityId);
+        return eventService.getEventsByCityId(cityId);
+    }
+
+    // Get all events by country ID
+    @GetMapping(value = "/country/{countryId}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public Flux<EventResponseModel> getEventsByCountryId(@PathVariable String countryId) {
+        log.info("Fetching all events by country ID: {}", countryId);
+        return eventService.getEventsByCountryId(countryId);
+    }
```

Comment:
Do these act like filters to filter events by cityId and countryId? If so, I think it would be better to use query parameters instead of creating separate endpoints.